burn-core/src/nn/rnn/lstm.rs

```diff
@@ -40,36 +46,31 @@ impl LstmConfig {
     pub fn init<B: Backend>(&self) -> Lstm<B> {
         let d_output = self.d_hidden;
 
-        let input_gate = gate_controller::GateController::new(
-            self.d_input,
-            d_output,
-            self.bias,
-            self.initializer.clone(),
-        );
-        let forget_gate = gate_controller::GateController::new(
-            self.d_input,
-            d_output,
-            self.bias,
-            self.initializer.clone(),
-        );
-        let output_gate = gate_controller::GateController::new(
-            self.d_input,
-            d_output,
-            self.bias,
-            self.initializer.clone(),
-        );
-        let cell_gate = gate_controller::GateController::new(
-            self.d_input,
-            d_output,
-            self.bias,
-            self.initializer.clone(),
-        );
+        let new_gate =
+            || GateController::new(self.d_input, d_output, self.bias, self.initializer.clone());
+
+        let (input_gate_bw, forget_gate_bw, output_gate_bw, cell_gate_bw);
+        if self.bidirectional {
+            (input_gate_bw, forget_gate_bw, output_gate_bw, cell_gate_bw) = (
+                Some(new_gate()),
+                Some(new_gate()),
+                Some(new_gate()),
+                Some(new_gate()),
+            );
+        } else {
+            (input_gate_bw, forget_gate_bw, output_gate_bw, cell_gate_bw) =
+                (None, None, None, None);
+        }
```

Comment:
Even if it's a boolean condition, I feel like a match might be easier to read, but using `new_gate` is a big win.